dvc/path_info.py

```diff
@@ -241,3 +241,79 @@ class CloudURLInfo(URLInfo):
     @property
     def path(self):
         return self._spath.lstrip("/")
+
+
+class HTTPURLInfo(URLInfo):
+    def __init__(self, url):
+        p = urlparse(url)
+        stripped = p._replace(params=None, query=None, fragment=None)
+        super().__init__(stripped.geturl())
+        self.params = p.params
+        self.query = p.query
+        self.fragment = p.fragment
+
+    @classmethod
+    def from_parts(
+        cls,
+        scheme=None,
+        host=None,
+        user=None,
+        port=None,
+        path="",
+        netloc=None,
+        params=None,
+        query=None,
+        fragment=None,
+    ):
+        assert bool(host) ^ bool(netloc)
+
+        if netloc is not None:
+            return cls(
+                "{}://{}{}{}{}{}".format(
+                    scheme,
+                    netloc,
+                    path,
+                    (";" + params) if params else "",
+                    ("?" + query) if query else "",
+                    ("#" + fragment) if fragment else "",
+                )
+            )
+
+        obj = cls.__new__(cls)
+        obj.fill_parts(scheme, host, user, port, path)
+        obj.params = params
+        obj.query = query
+        obj.fragment = fragment
```

Comment:
I didn't bother overriding `fill_parts` as it's not used publicly anywhere else

Reply:
Those use `from_parts` (which would this overridden method). The point is nothing else explicitly uses `fill_parts` directly so there's no need to override `fill_parts` for now.

Reply:
You keep saying don't need to override, but you override. Not sure I understand what you are up to here.

Reply:
I mean technically the correct way to do things would be:

```python
def fill_parts(self, scheme, host, user, port, path, params, query, fragment):
    super().fill_parts(self, scheme, host, user, port, path)
    self.params = params
    self.query = query
    self.fragment = fragment
```

but this isn't required right now as `fill_parts` is really a private method not used anywhere else.